Sources/Logging/MetadataProvider.swift

```diff
@@ -0,0 +1,112 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Logging API open source project
+//
+// Copyright (c) 2018-2019 Apple Inc. and the Swift Logging API project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of Swift Logging API project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+#if os(macOS) || os(iOS) || os(tvOS) || os(watchOS)
+import Darwin
+#elseif os(Windows)
+import CRT
+#elseif canImport(Glibc)
+import Glibc
+#elseif canImport(WASILibc)
+import WASILibc
+#else
+#error("Unsupported runtime")
+#endif
+
+#if compiler(>=5.6)
+@preconcurrency public protocol _SwiftLogSendable: Sendable {}
```

Comment:
Hmmm, I'll check but I think they need to be public -- it's for other libs to know this type is Sendable after all 🤔

Reply:
If I recall correctly this also works with an internal protocol. Maybe this changed but I tried to for one of our NIO repos. Somehow the compiler still transfers the Sendable constraint 😅

Reply:
I just tested this again. Since `Sendable` is only a marker protocol you don't have to make this `public`. It works without as well and propagates across modules.